app/src/main/java/com/example/diceroller/DiceViewAdapter.kt

```diff
@@ -1,49 +1,45 @@
 package com.example.diceroller
 
-import android.content.Context
 import android.view.LayoutInflater
 import android.view.ViewGroup
-import androidx.core.graphics.drawable.toDrawable
 import androidx.recyclerview.widget.RecyclerView
 import com.example.diceroller.databinding.ItemRolledDiceBinding
 import java.text.DateFormat
 
-class DiceViewAdapter (
-    private val items: List<Dice>
```

Comment:
❗ This is dangerous for performance. Passing items via the constructor and only through the constructor puts you in a position to generate a new Adapter every time the list of items changes.

I added a `setData()` function to your adapter that dynamically changes the list. This way you can use one adapter the entire time.

Reply:
it does make sense, I noticed looking at your code that you use `setData()` on your adapter and I was wondering if it's too hard to implement, but then I got dizzy looking at the way your ViewModel works so I decided to follow the documentation instead, gonna look more into this.